### PR TITLE
Minor refactoring about share servlet

### PR DESCRIPTION
**Pull request by ebyhr**

### CODE REVIEW

File: src/main/java/yanagishima/servlet/ShareCsvDownloadServlet.java [hunk 1]
```diff
@@ -1,50 +1,42 @@
...
-                String queryid = publishOptional.get().getQueryId();
-                Optional<String> encodeOptional = Optional.ofNullable(request.getParameter("encode"));
-                String header = request.getParameter("header");
-                if(header != null && header.equals("false")) {
-                    DownloadUtil.csvDownload(response, fileName, datasource, queryid, encodeOptional.orElse("UTF-8"), false);
-                } else {
-                    DownloadUtil.csvDownload(response, fileName, datasource, queryid, encodeOptional.orElse("UTF-8"), true);
-                }
-            });
+            boolean showHeader = Boolean.parseBoolean(request.getParameter("header"));
```
Comment (wyukawa) [inline src/main/java/yanagishima/servlet/ShareCsvDownloadServlet.java]: if ```header``` is null, ```showHeader``` must be true

File: src/main/java/yanagishima/servlet/ShareDownloadServlet.java [hunk 2]
```diff
@@ -1,50 +1,41 @@
...
-            });
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) {
+        Optional<String> publishId = Optional.ofNullable(request.getParameter("publish_id"));
+        if (publishId.isEmpty()) {
+            return;
+        }
+
+        db.single(Publish.class).where("publish_id=?", publishId.get()).execute().ifPresent(publish -> {
+            Optional<String> encode = Optional.ofNullable(request.getParameter("encode"));
+            boolean showHeader = Boolean.parseBoolean(request.getParameter("header"));
```
Comment (wyukawa) [inline src/main/java/yanagishima/servlet/ShareDownloadServlet.java]: ditto

File: src/main/java/yanagishima/servlet/ShareHistoryServlet.java [hunk 3]
```diff
@@ -7,71 +7,52 @@
...
-                HistoryUtil.createHistoryResult(retVal, yanagishimaConfig.getSelectLimit(), datasource, query, true);
-                Optional<Comment> commentOptional = db.single(Comment.class).where("datasource = ? and engine = ? and query_id = ?", datasource, query.getEngine(), queryid).execute();
-                if(commentOptional.isPresent()) {
-                    Comment comment = commentOptional.get();
-                    retVal.put("comment", comment);
-                } else {
-                    retVal.put("comment", null);
-                }
-            }
+            String publishId = request.getParameter("publish_id");
```
Comment (wyukawa) [inline src/main/java/yanagishima/servlet/ShareHistoryServlet.java]: I think ```HttpRequestUtil(request, "publish_id")``` is better

**Comment by ebyhr** [comment]
Updated.

**Pull request merged**